src/python/gem5/resources/suite.py

```diff
@@ -0,0 +1,123 @@
+# Copyright (c) 2023 The Regents of the University of California
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are
+# met: redistributions of source code must retain the above copyright
+# notice, this list of conditions and the following disclaimer;
+# redistributions in binary form must reproduce the above copyright
+# notice, this list of conditions and the following disclaimer in the
+# documentation and/or other materials provided with the distribution;
+# neither the name of the copyright holders nor the names of its
+# contributors may be used to endorse or promote products derived from
+# this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+# OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+from .workload import Workload
+
+"""
+struct suite: {
+    id: 'id',
+    resource_version: '1.0.0',
+    gem5_versions: ['21.0.0', '21.1.0'],
+    workloads: [
+        {
+            id: 'id',
+            resource_version: '1.0.0',
+            input_group: ['input_group_id', 'input_group_id2'],
+        },
+        {
+            id: 'id2',
+            resource_version: '1.0.0',
+            input_group: ['input_group_id', 'input_group_id3'],
+        }
+    ]
+
+}
+class suite:
+
+impl: iter, next(maybe) , len
+
+methods: list_input_groups, with_input_group
+"""
+
+from _m5 import core
+
+from typing import Dict, Any, List, Optional, Set
+
+
+class Suite:
+    def __init__(
+        self,
+        resource_id: str,
+        resource_version: str,
```

Comment:
We did that because we need the id and version to use obtain resources. We were thinking that rather than making a `suite.py` we would add `SuiteResource` to the `resource.py` like other resources so that class constructor will not neet id and version.